SWI-cpp2.cpp

```diff
@@ -982,8 +982,8 @@ _SWI_CPP2_CPP_check_rc(int, setenc(IOENC new_enc, IOENC *old_enc), Ssetenc(s_, n
 _SWI_CPP2_CPP_check_rc(int, setlocale(struct PL_locale *new_loc, struct PL_locale **old_loc), Ssetlocale(s_, new_loc, old_loc))
 _SWI_CPP2_CPP_check_rc(int, flush(), Sflush(s_))
 _SWI_CPP2_CPP_check_rc(int64_t, size(), Ssize(s_))
-_SWI_CPP2_CPP_check_rc(int, seek(long pos, int whence), Sseek(s_, pos, whence))
-_SWI_CPP2_CPP_check_rc(int, tell(), Stell(s_))
+_SWI_CPP2_CPP_check_rc(int, seek(int64_t pos, int whence), Sseek64(s_, pos, whence))
```

Comment:
I disagree with this change. It breaks the convention that the method name relates to the underlying PL_*() name.
And there's already PlStream::seek64() and PlStream::tell64(), which do what you want.
If  PL_seek() is changed to use "long" (or int64_t), then that would be fine. But otherwise, I would prefer a [[deprecated]] on PlStream::seek() and PlStream::tell()

Reply:
Loosing the direct mapping is indeed not nice.   On the other hand, introducing something that is deprecated when introduced is neither.    Except for Windows, this is practically doesn't change anything.  `long` is 64 bit on all modern platforms, so people may be tended to use tell() and seek() as it always did what it was meant to do except for the short interval when 32 bit systems were still common practice and files became bigger than 4Gb.   Now, 64 bit systems are common and all works as it did in the old days, except for Windows where `long` is only 32 bits.

I think this causes less surprises and makes more code run that didn't really catch up with all this 32/64 bit stuff correctly.

Still, as far as I'm concerned you are mostly in charge of the C++ api.  I'll accept a PR that changes things the way you want as long as it compiles silently on all our targets if the above does not convince you.

Reply:
FWIW, I agree. I don't know how to revert a change in git, could you take over? Otherwise, I am happy to start a new PR with "long" and the deprecation

Reply:
I see that it has already been merged, thanks